### PR TITLE
Move deduplication calculation from Logs component to reselect selector

### PR DESCRIPTION
**Pull request by dprokop**

### CODE REVIEW

File: public/app/features/explore/Logs.tsx [hunk 1]
```diff
@@ -142,7 +141,7 @@ export default class Logs extends PureComponent<Props, State> {
 
   onToggleLogLevel = (rawLevel: string, hiddenRawLevels: Set<string>) => {
```
Comment (hugohaggmark) [inline public/app/features/explore/Logs.tsx]: Minor: can ``rawLevel: string`` be removed from function signature?

File: public/app/features/explore/LogsContainer.tsx [hunk 2]
```diff
@@ -90,12 +108,29 @@ const selectItemUIState = (itemState: ExploreItemState) => {
     dedupStrategy,
   };
 };
+
+const logsSelector = (state: ExploreItemState) => state.logsResult;
```
Comment (hugohaggmark) [inline public/app/features/explore/LogsContainer.tsx]: All these selectors should be closer to the reducer (IMHO) and I'm not really convinced that we should have side effects in selectors. 
Couldn't all this be moved to the reducer and then you'll have to do this calculation for every action affected by this in the reducer right?

Comment (dprokop) [reply public/app/features/explore/LogsContainer.tsx]: The thing is that there are no side effects in the selector, only data processing. 

My thinking here is that deduped logs are function of `logsResults x dedupStrategy x hiddenLogsLevel` so there is no point in keeping them in store, if all the information required to calculate those is already there. Keeping them in store will introduce some more complexity to the reducers as well.I.e. we will have to dispatch dedup calc action on e.g. query run, while selector gives us this updates for free on input change.

Comment (hugohaggmark) [reply public/app/features/explore/LogsContainer.tsx]: Ok I understand, I'm not sure that I agree though 😄 , I think the PR looks good!

Comment (dprokop) [reply public/app/features/explore/LogsContainer.tsx]: Sure. Any benefits of keeping this calculated data in redux that comes to your mind?

Comment (hugohaggmark) [reply public/app/features/explore/LogsContainer.tsx]: It moves the processing away from the component. I'm more used to a pattern where the processing is done by the reducers and the components just connect to the processed state. But I think that is just my preference I don't think it really has any impact in this case.

Comment (dprokop) [reply public/app/features/explore/LogsContainer.tsx]: Well, I think it's my bad I have put selector in `LogsContainer` component. It definitely should belong to i.e. `explore/selectors.ts` as selectors are component agnostic and are used to retrieve data (derived or not) only.

Comment (hugohaggmark) [reply public/app/features/explore/LogsContainer.tsx]: I think moving the selectors there is an excellent suggestion!

File: public/app/features/explore/Logs.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/explore/Logs.tsx]: I don't want to remove it now as this is a part of `Graph` component API and I'm not sure what is the cost of removing it.

File: public/app/features/explore/LogsContainer.tsx (hunk 2 above)
Comment (dprokop) [reply public/app/features/explore/LogsContainer.tsx]: Moved in dd19ec3 :)

**Pull request merged**